lib/spack/spack/stage.py

```diff
@@ -545,7 +544,13 @@ def _add_to_root_stage(self):
                          '{stage}\n\tdestination : {destination}'.format(
                              stage=source_path, destination=destination_path
                          ))
-                shutil.move(os.path.realpath(source_path), destination_path)
+
+                src = os.path.realpath(source_path)
+
+                if os.path.isdir(src):
+                    copy_tree(src, destination_path)
```

Comment:
out of curiosity why doesn't `copy` handle this distinction?

Reply:
I guess I can't think of a reason why we couldn't redesign `copy` and `install` to support both files and directories. It's just a design choice. For comparison, Python's `shutil` library has separate `copy` and `copytree` methods. Python's `distutils` library also has separate `dir_util.copy_tree` and `file_util.copy_file` methods.